classes/class-p4-settings.php

```diff
@@ -0,0 +1,161 @@
+			}
+
+			$categories         = get_categories();
+
+			$categories_array[] = __( 'Select Category' ,'planet4-master-theme' ) ;
+			foreach ( $categories as $category ) {
+				$categories_array[$category->term_id] = __( $category->cat_name ,'planet4-master-theme' );
+			}
+
+			$this->fields = apply_filters( 'planet4_options', array(
+					array(
+						'name'    => __( 'Select act page', 'planet4-master-theme' ),
+						'id'      => 'select_act_page',
+						'type'    => 'select',
+						'options' => $pages_array
+					),
+
+					array(
+						'name'    => __( 'Select explore page', 'planet4-master-theme' ),
+						'id'      => 'select_explore_page',
+						'type'    => 'select',
+						'options' => $pages_array
+					),
+
+					array(
+						'name'    => __( 'Select category', 'planet4-master-theme' ),
+						'id'      => 'select_category',
+						'type'    => 'select',
+						'options' => $categories_array
+					)
+
+				) );
+			$this->hooks();
+		}
+
+		/**
+		 * Initiate our hooks
+		 */
+		public function hooks() {
+			add_action( 'admin_init', array( $this, 'init' ) );
+			add_action( 'admin_menu', array( $this, 'add_options_page' ) );
+		}
+
+		/**
+		 * Register our setting to WP
+		 */
+		public function init() {
+			register_setting( $this->key, $this->key );
+		}
+
+		/**
+		 * Add menu options page
+		 */
+		public function add_options_page() {
+			$this->options_page = add_options_page( $this->title, $this->title, 'manage_options', $this->key, array( $this, 'admin_page_display' ) );
+		}
+
+		/**
+		 * Admin page markup. Mostly handled by CMB2
+		 */
+		public function admin_page_display() {
+			?>
+			<div class="wrap <?php echo $this->key; ?>">
+				<h2><?php echo esc_html( get_admin_page_title() ); ?></h2>
+				<?php cmb2_metabox_form( $this->option_metabox(), $this->key ); ?>
+			</div>
+			<?php
+		}
+
+		/**
+		 * Defines the theme option metabox and field configuration
+		 * @return array
+		 */
+		public function option_metabox() {
+			return array(
+				'id'         => 'option_metabox',
+				'show_on'    => array( 'key' => 'options-page', 'value' => array( $this->key, ), ),
+				'show_names' => true,
+				'fields'     => $this->fields,
+			);
+		}
+
+		/**
+		 * Public getter method for retrieving protected/private variables
+		 * @param  string  $field Field to retrieve
+		 * @return mixed          Field value or exception is thrown
+		 */
+		public function __get( $field ) {
+			// Allowed fields to retrieve
+			if ( in_array( $field, array( 'key', 'fields', 'title', 'options_page' ), true ) ) {
+				return $this->{$field};
+			}
+
+			if ( 'option_metabox' === $field ) {
+				return $this->option_metabox();
+			}
+
+			throw new Exception( 'Invalid property: ' . $field );
+		}
+	}
+}
+
+/**
+ * Wrapper function around cmb2_get_option
+ * @param  string  $key Options array key
+ * @return mixed        Option value
+ */
+function planet4_get_option( $key = '' ) {
```

Comment:
In my opinion I don't think you really need this global function here. Better to let in here only the class.
I would call the cmb2_get_option() directly.

Reply:
If we use wrapper function planet4_get_option(), we will get more control over output of cmb2_get_option() function.

Reply:
@sagarsdeshmukh 
But why do you need cmb2_get_option? Does it do something that get_option() does not?
In my opinion wherever we have perfectly good and trusted WP functions that do the job, we should use those to avoid getting too tightly coupled with CMB2 and for future developers to be able to understand faster the code.